src/components/VideoLibraryModal.vue

```diff
@@ -0,0 +1,594 @@
+
+const videoStore = useVideoStore()
+
+const availableImages = [
+  {
+    src: 'https://upload.wikimedia.org/wikipedia/commons/8/8a/Discover_Scuba_Diving_--_St._Croix%2C_US_Virgin_Islands.jpg',
+    isSelected: false,
+    title: 'Cockpit (Mar 12, 2024 - 16꞉55꞉51 GMT-3) #76c023ca.png',
+  },
+  {
+    src: 'https://bluerobotics.com/wp-content/uploads/2016/06/BlueROV2.jpg',
+    isSelected: true,
+    title: 'Cockpit (Mar 12, 2024 - 16꞉55꞉51 GMT-3) #76c023ca.png',
+  },
+]
+
+const dummySelectedImage = availableImages.find((video) => video.isSelected)
+
+const props = defineProps({
+  openModal: Boolean,
+})
+
+const emits = defineEmits(['update:openModal'])
+
+// List available videos and telemetry logs to be downloaded
+/* eslint-disable jsdoc/require-jsdoc  */
+interface VideoStorageFile {
+  filename: string
+  size: number
+  url: string
+}
+
+/* eslint-enable jsdoc/require-jsdoc  */
+const availableVideosAndLogs = ref<VideoStorageFile[]>([])
+const isVisible = ref(props.openModal)
+const selectedVideos = ref<VideoStorageFile[]>([])
+const selectedImage = ref(dummySelectedImage)
+const currentTab = ref('videos')
+const temporaryDbSize = ref(0)
+const openSnackbar = ref(false)
+const snackbarMessage = ref('')
+const longPressTimeoutId = ref<number | null>(null)
+const isLongPress = ref(false)
+
+// Track the blob URLs to revoke them when the modal is closed
+const blobURLs = ref<string[]>([])
+
+const menuButtons = [
+  { name: 'Videos', icon: 'mdi-video-outline', selected: true, enable: true, tooltip: '' },
+  { name: 'Pictures', icon: 'mdi-image-outline', selected: false, enable: false, tooltip: 'Coming soon' },
+]
+
+const fileActionButtons = [
+  {
+    name: 'Delete',
+    icon: 'mdi-delete-outline',
+    size: 22,
+    confirmAction: true,
+    enable: true,
+    action: () => discardAndUpdateDB(selectedVideos.value.map((video: VideoStorageFile) => video.filename)),
+  },
+  {
+    name: 'Download',
+    icon: 'mdi-tray-arrow-down',
+    size: 28,
+    confirmAction: false,
+    enable: true,
+    action: () => downloadAndUpdateDB(selectedVideos.value.map((video: VideoStorageFile) => video.filename)),
+  },
+]
+
+const closeModal = (): void => {
+  isVisible.value = false
+  emits('update:openModal', false)
+  currentTab.value = 'videos'
+  blobURLs.value.forEach((url) => URL.revokeObjectURL(url))
+  blobURLs.value = []
+}
+
+const parseDateFromTitle = (title: string): string => {
+  const dateRegex = /\(([^)]+)\)/
+  const dateMatch = title.match(dateRegex)
+  return dateMatch ? dateMatch[1] : ''
+}
+
+// Long press for multiple video selection
+const startLongPressCheck = (video: VideoStorageFile): void => {
+  isLongPress.value = false
+  if (longPressTimeoutId.value !== null) {
+    clearTimeout(longPressTimeoutId.value)
+  }
+  longPressTimeoutId.value = window.setTimeout(() => {
+    isLongPress.value = true
+    handleSelectionForLongPress(video)
+  }, 500)
+}
+
+// Selects video on long press
```

Comment:
This is another example where the comment is an duplication of the code.
For comments, I usually recommend this: https://www.youtube.com/watch?v=NLebZ3XT92E a must watch